lib/govuk_sidekiq/error_handling.rb

```diff
@@ -0,0 +1,33 @@
+module GovukSidekiq
+  module ErrorHandling
+    class RetryWorker < StandardException
+      def initialize(original_exception)
+        @original_exception = original_exception
+      end
+
+      attr_reader :original_exception
+    end
+
+    class Middleware
+      def call(worker, job, queue)
+        yield
+      rescue StandardException => e
+        handle_exception(job, e)
+      end
+
+      def handle_exception(job, e)
+        if last_try?(job)
+          raise e
+        else
+          raise RetryWorker.new(e)
+        end
+      end
+
+      def last_try?(job)
+        max_retries = job["retries"] || Sidekiq::Middleware::Server::RetryJobs::DEFAULT_MAX_RETRY_ATTEMPTS
+        retry_count = job["retry_count"] || 0
+        !job["retry"] || retry_count == max_retries - 1
```

Comment:
are we at any risk of `job["retry"]` being nil by default and thus stopping retries?

Reply:
If it's `nil`, `job["retry"]` should be `true` meaning `last_try?` should be `true` and then the ordinary exception will be raised. I've actually refactored this logic to make it a bit clearer.